src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -351,6 +351,8 @@ protected IntPtr LoadUnmanagedDllFromPath(string unmanagedDllPath)
             {
                 throw new ArgumentException(SR.Argument_AbsolutePathRequired, nameof(unmanagedDllPath));
             }
+            
+            unmanagedDllPath = PathInternal.RemoveRelativeSegments(unmanagedDllPath, PathInternal.GetRootLength(unmanagedDllPath));
```

Comment:
I really don't understand this function call. The issue seems to be related to un-normalized path delimiters. Shouldn't there be a place deeper in the loader that handles that?

Reply:
The rest of the paths already normalize. I'll see if I can move this lower in the stack.

Reply:
Moved it down the stack.